packages/grid/_modules_/grid/hooks/features/focus/useGridFocus.ts

```diff
@@ -33,11 +33,9 @@ export const useGridFocus = (apiRef: GridApiRef): void => {
           focus: { cell: { id, field }, columnHeader: null },
         };
       });
-      // TODO replace with constant
-      apiRef.current.publishEvent('cellFocusChange');
```

Comment:
It should have been a constant, probably something that was missed in a review. 
The idea was to publish an event when the grid focus state changes so ppl can subscribe to state change. Now you can use 'cellFocus' to know which cell will be next to get the focus but the grid state won't have changed at that stage. So it's slightly different

Reply:
> The idea was to publish an event when the grid focus state changes so ppl can subscribe to state change.

This seems to resonate with the potential confusion about what these events are. This is not clear to me. Are the events coming from user events, regardless of if the data grid commits them or not. For instance, a GRID_PAGE_CHANGE fires as long as the user clicks to change the current state, and regardless of if the outcome changing the state or not. Or does it trigger once a state change is persisted? 

Here, seems `cellFocusChange` only fire 50% of the time, it seems to be more of a legacy logic we broke along the way, but for which nobody complained about. So I would 👍 for removing it and see if developers miss it.

Reply:
For these events that are fired after a state update, we could add a reason to `setState` and developers can listen to GRID_STATE_CHANGE. We could go further and extend the new useControlState hook to allow to subscribe to updates in just a part of the state.

Reply:
Being able to listen to state changes would help making sure that code like #1815 only runs when the Grid is ready enough.

But on the other hand, I think this pattern could easily be overused. A lot of events would have their equivalent with `GRID_STATE_CHANGE`.

Reply:
> We could go further and extend the new useControlState hook to allow to subscribe to updates in just a part of the state.

@m4theushw It sounds close to the point 2. of https://github.com/mui-org/material-ui-x/issues/1507#issuecomment-828485605.

@flaviendelangle Agree, I think that the event should fire to signal that the action was initiated, not to signal that it has been persisted, usually, developers don't need to know. For instance, we could fire the event on a page change, event if the state is controlled and the change, was therefore not committed. So it's really to signal the end-user intent, not to signal the outcome.